packages/core/src/metrics/metric-summary.ts

```diff
@@ -0,0 +1,87 @@
+import type { MeasurementUnit, Span } from '@sentry/types';
+import type { MetricSummary } from '@sentry/types';
+import type { Primitive } from '@sentry/types';
+import { dropUndefinedKeys } from '@sentry/utils';
+import { getActiveSpan } from '../tracing';
+import type { MetricType } from './types';
+
+/**
+ * key: bucketKey
+ * value: [exportKey, MetricSummary]
+ */
+type MetricSummaryStorage = Map<string, [string, MetricSummary]>;
+
+let SPAN_METRIC_SUMMARY: WeakMap<Span, MetricSummaryStorage> | undefined;
+
+function getMetricStorageForSpan(span: Span): MetricSummaryStorage | undefined {
+  return SPAN_METRIC_SUMMARY ? SPAN_METRIC_SUMMARY.get(span) : undefined;
+}
+
+/**
+ * Fetches the metric summary if it exists for the passed span
+ */
+export function getMetricSummaryJsonForSpan(span: Span): Record<string, Array<MetricSummary>> | undefined {
+  const storage = getMetricStorageForSpan(span);
+
+  if (!storage) {
+    return undefined;
+  }
+  const output: Record<string, Array<MetricSummary>> = {};
+
+  for (const [, [exportKey, summary]] of storage) {
+    output[exportKey] = [...(output[exportKey] || []), dropUndefinedKeys(summary)];
```

Comment:
spreading `output[exportKey]` is expensive in a loop, let's just mutate directly.

```suggestion
    if (!output[exportKey]) {
      output[exportKey] = [];
    }
    
    output[exportKey].push(dropUndefinedKeys(summary));
```